workshop_app/views.py

```diff
@@ -44,23 +44,39 @@ def is_instructor(user):
     return user.groups.filter(name='instructor').exists()
 
 
-def get_landing_page(user):
-    # For now, landing pages of both instructor and coordinator are same
+def instructor_only(func):
+    def inner1(*args, **kwargs):
+        if is_instructor(args[0].user):
+            return func(*args, **kwargs)
+        return redirect(reverse('workshop_status_coordinator'))
+
+    return inner1
+
+
+def coordinator_only(func):
+    def inner1(*args, **kwargs):
+        if not is_instructor(args[0].user):
+            return func(*args, **kwargs)
+        return redirect(reverse('workshop_status_instructor'))
+
+    return inner1
+
+
+def redirect_to_landing_page(user):
     if is_instructor(user):
         return reverse('workshop_status_instructor')
     return reverse('workshop_status_coordinator')
 
 
 # View functions
 
+@login_required()
+@instructor_only
+@coordinator_only
 def index(request):
     """Landing Page : Redirect to login page if not logged in
                       Redirect to respective landing page according to position"""
-    user = request.user
-    if user.is_authenticated and is_email_checked(user):
-        return redirect(get_landing_page(user))
-
-    return redirect('/login/')
+    pass
```

Comment:
why have this function if it is not doing anything?

Reply:
This function redirects users who come to '/' route to their respective landing pages. As CMS app is not yet merged, not having this function will cause a 404 error on '/' route.